### DISCUSSION TITLE
how to add npm packages like js-yaml when writing our own javascript files

### DISCUSSION THREAD

**Discussion by stephenzwj** [alshedivat/al-folio#898] [Q&A] [answered]
Hello, team, can you please enlighten me on how to use external packages? I want to use `js-yaml` in my customized javascript. But I did not manage to do it as the ` require ` package needed to be loaded first.

> const yaml = require('js-yaml');
> const fs = require('fs');
> const urls = yaml.load(fs.readFileSync("_data/venues.yml", 'utf8'));
> const urls2 = yaml.load(fs.readFileSync("_data/coauthors.yml", 'utf8'));
> for (const [key, value] of Object.entries(urls2)) {
>     urls[key] = value[0];
>   }
> console.log(urls);

**Comment by rohandebsarkar** [comment, marked as answer]
You can try adding a script tag with the required script to load the script before calling it. For this particular case, this might work (not tested). `<script src="https://cdn.jsdelivr.net/npm/js-yaml@4/dist/js-yaml.min.js"></script>`

**Reply by stephenzwj** [reply]
Thank you so much for your reply. Where shall we put this piece of code so that it can be effective for all pages? I am putting my js code into the common.js file

**Reply by rohandebsarkar** [reply]
Try putting it into `_includes/head.html`.

**Reply by stephenzwj** [reply]
Thank you so much for your answer. I am having another issue with the fs package as I need to use it to do `fs.readFileSync("_data/venues.yml", 'utf8')`. Do you have a suggestion for this? Thank you so much.
